pulp/apis/sas_api.py

```diff
@@ -0,0 +1,623 @@
+                # TODO: Check whether there is better implementation than zip().
+                values = dict(zip(primal_out['_VAR_'], primal_out['_VALUE_']))
+                rc = dict(zip(primal_out['_VAR_'], primal_out['_R_COST_']))
+                lp.assignVarsVals(values)
+                lp.assignVarsDj(rc)
+
+                prices = dict(zip(dual_out['_ROW_'], dual_out['_VALUE_']))
+                slacks = dict(zip(dual_out['_ROW_'], dual_out['_ACTIVITY_']))
+                lp.assignConsPi(prices)
+                lp.assignConsSlack(slacks, activity=True)
+            else:
+                # Convert primal out data set to variable dictionary
+                # Use pandas functions for efficiency
+                values = dict(zip(primal_out['_VAR_'], primal_out['_VALUE_']))
+                lp.assignVarsVals(values)
+            lp.assignStatus(status)
+            return status
+
+
+class SASCAS(LpSolver):
+    name = "SASCAS"
+
+    try:
+        global swat
+        import swat
+
+    except ImportError:
+
+        def available(self):
+            """True if SASCAS is available."""
+            return False
+
+        def actualSolve(self, lp, callback=None):
+            """Solves a well-formulated lp problem."""
+            raise PulpSolverError("SASCAS : Not Available")
+
+    else:
+        def __init__(
+            self,
+            mip=True,
+            msg=True,
+            warmStart=False,
+            cas_options={},
+            **solverParams,
+        ):
+            """
+            :param bool mip: if False, assume LP even if integer variables
+            :param bool msg: if False, no log is shown
+            :param bool warmStart: if False, no warmstart or initial primal solution provided.
+            """
+            LpSolver.__init__(
+                self,
+                mip=mip,
+                msg=msg,
+            )
+            self._sas = swat
+            self.warmStart=warmStart
+            self.cas_options=cas_options
+            # Only named options are allowed in SAS solvers.
+            self.solverOptions = {key:value for key, value in solverParams.items()}
+        def available(self):
+            """True if SAS94 is available."""
+            return True
+
+
+        def toDict(self):
+            data = dict(solver=self.name)
+            for k in ["mip", "msg", "warmStart", "cas_options"]:
+                try:
+                    data[k] = getattr(self, k)
+                except AttributeError:
+                    pass
+            data.update(self.solverOptions)
+            return data
+
+        def _create_tmp_files(self, name, *args):
+
+            return (f"{name}-pulp.{n}" for n in args)
+
+        def _delete_tmp_files(self, *args):
+            for file in args:
+                try:
+                    os.remove(file)
+                except FileNotFoundError:
+                    pass
+
+        def _write_sol(self, filename, vs):
```

Comment:
there are many methods that are common between the two solvers. Could we just inherit one from the other and get all those methods?